src/Control/Monad/Free.hs

```diff
@@ -53,14 +51,24 @@ import Control.Monad.State.Class
 import Control.Monad.Error.Class
 import Control.Monad.Cont.Class
 import Data.Functor.Bind
+import Data.Functor.Classes
 import Data.Foldable
 import Data.Profunctor
 import Data.Traversable
 import Data.Semigroup.Foldable
 import Data.Semigroup.Traversable
 import Data.Data
 import Prelude hiding (foldr)
-import Prelude.Extras
+
+import qualified Prelude.Extras as PE
```

Comment:
If we leave it in place, then this change is backwards-compatible. That might be good idea, as then there is not hurry with dependencies.

OTOH, if contexts of `Eq (Free f a)` instances are changed to use `Eq1` from `base`, then there is less reason to have `prelude-extras` lying around.

I'd say, it's up to @ekmett to decide how to proceed.

--- 

FWIW: I'd change the `Eq`/`Ord`/`Show`/`Read` instances, with `cabal new-build` it's pretty easy to update reverse dependencies, and I could deal with the ripples.